cloudbuild/python-scripts/parse_logs.py

```diff
@@ -0,0 +1,513 @@
+from google.cloud import storage
+
+
+def get_bq_query_result_row_count(client_project_name, query):
+    client = bigquery.Client(project=client_project_name)
+    query_job = client.query(query)
+    query_result = query_job.result()
+    records_read = query_result.total_rows
+    return records_read
+
+
+def get_bq_table_row_count(
+    client_project_name, project_name, dataset_name, table_name, query
+):
+    """Method to get count of rows in a BigQuery Table.
+
+    Args:
+      client_project_name: Name of the project to form the BQ Client
+      project_name: Project ID that contains the table
+      dataset_name: Name of Dataset containing the table
+      table_name: Table Name
+      query: Query string [if any] that was provided to the connector. Incase
+        non-empty, The method executes this query and returns the number of rows
+        obtained.
+
+    Returns:
+      Count of rows in the provided `project_name.dataset_name.table_name` table
+    """
+    if query:
+        return get_bq_query_result_row_count(client_project_name, query)
+    dataset_ref = bigquery.DatasetReference(
+        project=project_name, dataset_id=dataset_name
+    )
+    table_ref = bigquery.TableReference(
+        dataset_ref=dataset_ref, table_id=table_name
+    )
+    client = bigquery.Client(project=client_project_name)
+    table = client.get_table(table=table_ref)
+    row_count = table.num_rows
+    return row_count
+
+
+def extract_metric(logs_as_string, metric_string, end_of_metric_string):
+    """Method to extract the metric value from logs downloaded as a string.
+
+    Args:
+      logs_as_string: Yarn application logs downloaded as a string.
+      metric_string: The string to be found
+      end_of_metric_string: Delimiter at the end of the value.
+
+    Returns:
+      Sum of values of the metric obtained in a file. As there can be 0 or more
+      occurrences of metric in a file
+    """
+    total_metric_sum_in_blob = 0
+    # Keep on finding the metric value as there can be
+    # 1 or more outputs in a log file.
+
+    # The logs are of the format
+    # Number of records read: <value> ;
+    # Here, "Number of records read: " is the metric string
+    # and ";" is the end_of_metric_string.
+    # We find the smallest possible string that matches.
+
+    metric_pattern = r'{}\s*(.*?)\s*{}'.format(
+        re.escape(metric_string), re.escape(end_of_metric_string)
+    )
+    metric_pattern = re.compile(metric_pattern)
+    matches = metric_pattern.finditer(logs_as_string)
+    records_read = [int(record_read.group(1).strip()) for record_read in matches]
+    total_metric_sum_in_blob += sum(records_read)
+    return total_metric_sum_in_blob
+
+
+def check_query_correctness(gcs_log_object, logs_as_string):
+    """Checks the correctness of query results obtained in the logs.
+
+    This is a hardcoded check, done by checking if the records obtained
+    from a "filter" query contains only the desired (filtered) values.
+    Args:
```

Comment:
Enter a blank line between these lines for consistency with other methods

Reply:
Fixed.